lametro/utils.py

```diff
@@ -43,33 +43,57 @@ def parse_subject(text):
-        previous_meeting = found_events.filter(start_time__gte=time_ago).last().get_previous_by_start_time()
+def calculate_current_meetings(found_events, now=datetime.now(app_timezone)):
+    '''
+    Accept a queryset of events that started in the last six hours, or start
+    in the next five minutes, and determine which is in progress, by checking
+    Legistar for an "In Progress" indicator.
+
+    If there are meetings scheduled concurrently, and one is a board meeting,
+    show the board meeting first.
+
+    Based on a spreadsheet of average meeting times from Metro, we assume
+    meetings last at least 55 minutes. If any found events started less than
+    55 minutes ago, display them as current if the previous meeting has
+    ended. Otherwise, check Legistar for the "In Progress" indicator.
+
+    This method should always return a queryset.
+    '''
+    fifty_five_minutes_ago = now - timedelta(minutes=55)
+
+    # '.annotate' adds a field called 'val', which contains a boolean – we order
+    # in reverse, since False comes before True, to show board meetings first.
+    found_events = found_events.annotate(val=RawSQL("name like %s", ('%Board Meeting%',)))\
+                               .order_by('-val')
+
+    earliest_start = found_events.earliest('start_time').start_time
+    latest_start = found_events.latest('start_time').start_time
+
+    if found_events.filter(start_time__gte=fifty_five_minutes_ago):
+        previous_meeting = found_events.filter(start_time__gte=fifty_five_minutes_ago)\
+                                       .last()\
+                                       .get_previous_by_start_time()
+
         if legistar_meeting_progress(previous_meeting):
             return LAMetroEvent.objects.filter(ocd_id=previous_meeting.ocd_id)
 
-        return found_events.filter(start_time__gte=time_ago)  
-    elif earliest_start == latest_start:  
-        # The IF statement handles the below cases:
-        # (1) found_events includes just one event object. Example: the first committee meeting of the day - 9:00 am on 01/18/2018
-        # (2) found_events includes multiple events that all happen at the same time (though in reality, they occur one-after-the-other). Example: the events at 9:00 am on 11/30/2017
+        return found_events.filter(start_time__gte=fifty_five_minutes_ago)
+
+    elif earliest_start == latest_start:
+        # There is one event object, or there are multiple events scheduled to
+        # happen at the same time.
+        #
+        # n.b., In reality, concurrently scheduled events happen one after the
+        # other, but we want the current meeting display to line up with the
+        # scheduled events (i.e., if it's 9:15, show all of the events slated
+        # for 9, even if only one is technically in progress).
         for event in found_events:
             if legistar_meeting_progress(event):
                 return found_events
-    else: 
+
+    else:
```

Comment:
The `if` here seems to be if there are multiple events that started in the last 55 minutes, the `elif` seems to be, as you descibe, there are multiple events scheduled to happen at the same time (or if there is only one). What is the `else` for? Are we ever going to follow this path or is it just defensive programming?

Reply:
it happens if you have multiple events scheduled in a day, that don't start at the same time. so like, if it's 2 p.m., and there are 9 a.m. and 12 p.m. events, they'll both come through as found events, and we test each one to see which is happening now (if any). 

i think there is a simpler way to write this logic, but i'm not sure i want to do that, in this pr... do you have any immediate thoughts?